app/(authenticated)/layout.tsx

```diff
@@ -1,29 +1,41 @@
-import { redirect } from 'next/navigation';
+'use client';
+
+import { redirect, usePathname } from 'next/navigation';
 import Main from '../../components/layouts/main';
 import Navbar from '../../components/layouts/navbar';
 import Sidebar from '../../components/layouts/sidebar';
 import { PAGE_ROUTES } from '@/utils/constants/common-constants';
-import { getServerSession } from 'next-auth';
-import { authOptions } from '@/lib/next-auth/auth';
+import { useSession } from 'next-auth/react';
+import { useEffect, useState } from 'react';
 
-export default async function AuthenticatedLayout({
-  children,
-}: {
-  children: React.ReactNode;
-}) {
-  const session = await getServerSession(authOptions);
+export default function AuthenticatedLayout({ children }: { children: React.ReactNode }) {
+  const [loading, setloading] = useState(true);
+  const path = usePathname();
+  const session = useSession();
+  const userRole = session?.data?.user?.user_type;
 
-  if (!session) {
-    redirect(PAGE_ROUTES.Signin);
-  }
+  useEffect(() => {
+    if (session?.status !== 'loading') {
+      setloading(false);
+    }
+    if (!session?.data && session?.status === 'unauthenticated') {
+      redirect(PAGE_ROUTES.Signin);
+    }
+    if (path === '/employee-list' && userRole === 'executive') {
+      redirect(PAGE_ROUTES.Dashboard);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [path, session, userRole]);
 
-  return (
-    <div className='flex h-screen bg-[#F7F7FB] overflow-hidden'>
-      <Sidebar />
-      <div className='flex-1 overflow-y-auto'>
-        <Navbar />
-        <Main>{children}</Main>
+  if (session?.data && !loading) {
```

Comment:
What should we show if this condition is `false`?  While it is loading, we can show a infinity loader in the middle of a blank (white) page.

Reply:
ok vaiya